packages/core/src/config-engine/config-engine.ts

```diff
@@ -405,32 +382,22 @@ export class DmnoService {
   }
 
   overrideSources: Array<OverrideSource> = [];
-  async loadOverrideFiles() {
-    // TODO: this is not at all optimized for speed...
-    // particularly it is doing a check on if the file is gitignored
-    // and if we are loading not in dev mode, we may just want to load files that will be applied
-    const dotEnvFiles = await loadServiceDotEnvFiles(this.path, { onlyLoadDmnoFolder: true });
-
-    // TODO: support other formats (yaml, toml, json) - probably should all be through a plugin system
-
-    // loads multiple override files, in order from more specific to least
-    // .env.{ENV}.local
-    // .env.local
-    // .env.{ENV}
-    // .env
-
-    this.overrideSources = _.map(dotEnvFiles, (dotEnvFile) => {
-      return new OverrideSource(
-        '.env file',
-        dotEnvFile.fileName,
-        'simple-icons:dotenv',
-        dotEnvFile.envObj,
-        // TODO: specific env overrides are being enabled based on process.env.NODE_ENV
-        // we probably want to be smarter about how _that_ gets resolved first
-        // and store it at the workspace level or something...?
-        !dotEnvFile.applyForEnv || dotEnvFile.applyForEnv === process.env.NODE_ENV,
-      );
+  async loadOverrides() {
+    this.overrideSources = [];
+
+    const overridePlugins = this.rawConfig?.overrides
+      ? _.compact(this.rawConfig?.overrides)
+      // default behaviour is to load overrides from process.env and dotenv files
+      : [processEnvOverrideLoader(), dotEnvFileOverrideLoader()];
```

Comment:
we'll need to update `/docs/guides/env-files/#setting-overrides-via-env-files` to denote any changes to order/precedence when using these override plugins

Reply:
💯  there's a whole overrides guide that needs to be written, and a good callout to make sure the .env file guide (if it still exists) needs to mention it. But luckily this is implemented so that the default behaviour is nothing changes.